src/rust/ru.mangaonelove/src/parser.rs

```diff
@@ -0,0 +1,364 @@
+		status,
+		nsfw,
+		viewer,
+	})
+}
+
+pub fn parse_chapters(html: &WNode, manga_id: &str) -> Option<Vec<Chapter>> {
+	let manga_chapters_holder_node =
+		html.select_one("div.c-page-content div#manga-chapters-holder")?;
+
+	let data_id = manga_chapters_holder_node.attr("data-id")?;
+
+	let real_manga_chapters_holder_node = post(
+		&format!("{BASE_URL}/wp-admin/admin-ajax.php"),
+		&format!("action=manga_get_chapters&manga={data_id}"),
+		&[
+			("X-Requested-With", "XMLHttpRequest"),
+			("Referer", &format!("{}", get_manga_url(manga_id))),
+		],
+	)
+	.ok()?;
+
+	let chapter_nodes = real_manga_chapters_holder_node.select("ul > li.wp-manga-chapter");
+
+	let abs = |l, r| {
+		if l > r {
+			l - r
+		} else {
+			r - l
+		}
+	};
+
+	let chapters = chapter_nodes
+		.into_iter()
+		.enumerate()
+		.filter_map(|(idx, chapter_node)| {
+			let url_node = chapter_node.select_one("a")?;
+			let url = url_node.attr("href")?;
+			let id = url
+				.substring_after(&format!("{}/", get_manga_url(manga_id)))?
+				.trim_end_matches('/')
+				.to_string();
+			let title = url_node.text();
+
+			let chapter = {
+				let approx_chapter = (idx + 1) as f32;
+				let mut possible_chapters: Vec<_> = title
+					.split_whitespace()
+					.filter_map(|word| word.parse::<f32>().ok())
+					.collect();
+				match possible_chapters[..] {
+					[] => approx_chapter,
+					[chap] => chap,
+					_ => {
+						possible_chapters.sort_by(|&l, &r| {
+							abs(l, approx_chapter)
+								.partial_cmp(&abs(r, approx_chapter))
+								.unwrap()
+						});
+						possible_chapters.first().cloned().unwrap_or(approx_chapter)
+					}
+				}
+			};
+
+			let extract_multiplier = |metric_str: &&str| {
+				if metric_str.starts_with("сек") {
+					Some(1)
+				} else if metric_str.starts_with("мин") {
+					Some(60)
+				} else if metric_str.starts_with("час") {
+					Some(60 * 60)
+				} else if metric_str.starts_with("дн") {
+					Some(24 * 60 * 60)
+				} else {
+					None
+				}
+			};
+
+			let date_updated = {
+				let release_date_node = chapter_node.select_one("span.chapter-release-date")?;
+				let normal_release_date = release_date_node.select_one("i").map(|i_node| {
+					StringRef::from(&i_node.text()).as_date("dd-MM-yyyy", None, None)
+				});
+
+				let ago_extractor = || {
+					release_date_node
+						.select_one("a")
+						.and_then(|a| a.attr("title"))
+						.and_then(|updated_text| {
+							if !updated_text.ends_with("ago") {
+								return None;
+							}
+							let spl: Vec<_> = updated_text.split_whitespace().collect();
+							let count = spl
+								.first()
+								.and_then(|count_str| count_str.parse::<f64>().ok())?;
+
+							let metric_mult = spl.get(1).and_then(extract_multiplier)?;
+
+							Some(current_date() - count * (metric_mult as f64))
+						})
+						.unwrap_or(0f64)
+				};
+
+				normal_release_date.unwrap_or_else(ago_extractor)
+			};
+
+			Some(Chapter {
+				id,
+				title,
+				volume: 1f32, // TODO: implement proper parsing
```

Comment:
why hardcode volume 1 instead of having no volume?

Reply:
Didn't know that default value meant "no volume info"
Fixed